nilearn/plotting/html_connectome.py

```diff
@@ -17,6 +18,34 @@ class ConnectomeView(HTMLDocument):
     pass
 
 
+def _encode_coordinates(coords, prefix):
+    """
+    Transform a 2D-array of 3D data (x, y, z) into a dict of base64 values
+
+    Parameters
+    ----------
+    coords : ndarray, shape=(n_nodes, 3)
+        The coordinates of the nodes in MNI space.
+
+    prefix : str
+        Prefix for the key value in the returned dict. Schema is {prefix}{x|y|z}
+
+    Returns
+    -------
+    coordinates : dict
+        Dictionary containing base64 values for each axis
+    """
+    coordinates = {}
+
+    coords = np.asarray(coords, dtype='<f4')
+    marker_x, marker_y, marker_z = coords.T
+    for coord, cname in [(marker_x, "x"), (marker_y, "y"), (marker_z, "z")]:
+        coordinates["{}{}".format(prefix, cname)] = encode(
+            np.asarray(coord, dtype='<f4'))
+
+    return coordinates
+
+
 def _prepare_line(edges, nodes):
```

Comment:
```suggestion
def _prepare_line(edges, nodes):
    """prepare a plotly scatter3d line plot so that a set of disconnected edges
    can be drawn as a single line.

    `edges` are values associated with each edge (that get mapped to colors
    through a colorscale). `nodes` are pairs of (source, target) node indices
    for each edge.

    the color of a line segment in plotly is a mixture of the colors associated
    with the points it connects. Moreover, segments that begin or end at a
    point whose value is `null` are not drawn.

    given edges = [eab, ecd, eef] and nodes = [(a, b), (c, d), (e, f)], this
    function returns:

        path_edges: eab eab   0 ecd ecd   0 eef eef   0
        path_nodes:   a   b   0   c   d   0   e   f   0

    moreover the javascript code replaces every third element (the '0' in the
    lists above) with `null`, so only the a-b, c-d, and e-f segments will get
    plotted, and their colors are correct because both their start and end
    points are associated with the same value.

    """
```

Reply:
I think the role of this function made it harder to get started on this PR, was
that the case? If so, we can add a comment like the one above